src/api/java/appeng/api/util/AECableType.java

```diff
@@ -64,4 +64,97 @@
 			DENSE_SMART
 	};
 
+	private final AECableVariant variant;
+	private final AECableSize size;
+
+	private AECableType( AECableVariant variant, AECableSize size )
+	{
+		this.variant = variant;
+		this.size = size;
+	}
+
+	public AECableSize size()
+	{
+		return size;
+	}
+
+	public AECableVariant variant()
+	{
+		return variant;
+	}
+
+	public boolean isValid()
+	{
+		return this.variant != AECableVariant.NONE && this.size != AECableSize.NONE;
+	}
+
+	public boolean isDense()
+	{
+		return this.size == AECableSize.DENSE;
+	}
+
+	public boolean isSmart()
+	{
+		return this.variant == AECableVariant.SMART;
+	}
+
+	public static AECableType min( AECableType a, AECableType b )
+	{
+		final AECableVariant v = AECableVariant.min( a.variant(), b.variant() );
+		final AECableSize s = AECableSize.min( a.size(), b.size() );
+
+		return AECableType.from( v, s );
+	}
+
+	public static AECableType max( AECableType a, AECableType b )
+	{
+		final AECableVariant v = AECableVariant.max( a.variant(), b.variant() );
+		final AECableSize s = AECableSize.max( a.size(), b.size() );
+
+		return AECableType.from( v, s );
+	}
+
+	private static AECableType from( AECableVariant variant, AECableSize size )
+	{
+		switch( variant )
+		{
+			case GLASS:
+				switch( size )
+				{
+					case THIN:
+						return GLASS;
+					default:
+						break;
+				}
+
+				break;
+			case COVERED:
+				switch( size )
+				{
+					case NORMAL:
+						return COVERED;
+					case DENSE:
+						return DENSE_COVERED;
+					default:
+						break;
+				}
+
+				break;
+			case SMART:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 10 to at most 5, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1151)